Add tests

petro2061 opened the pull request:
This task added classes to test the services, repositories, and controllers for the book and category entities. To test the service level, the mock-up object approach was used. Testing of the storage layer takes place in test containers to simulate real databases. The controller level is tested using integration tests.
![test-with-coverage](https://github.com/user-attachments/assets/7d2ddcb5-4976-4a0a-8edd-89d103a761b7)

petro2061 committed:
started writing book service tests

petro2061 committed:
Merge branch 'refs/heads/add-docker' into add-tests

petro2061 committed:
added correct import exception class

petro2061 committed:
added new test methods

petro2061 committed:
fixed methods implementation

petro2061 committed:
added tests for CategoryServiceImpl class

petro2061 committed:
added dependencies for testing repository layer

petro2061 committed:
added properties for tests

petro2061 committed:
implemented class for test in testcontainers

petro2061 committed:
added sql scripts for repository test

petro2061 committed:
added test for book repository

petro2061 committed:
added new dependency

petro2061 committed:
added BookControllerTest class

petro2061 committed:
added new test method

petro2061 committed:
added assertj dependency

petro2061 committed:
changed configuration

petro2061 committed:
added transaction annotation over required methods

petro2061 committed:
changed jdk version

petro2061 committed:
changed mysql version

petro2061 committed:
fixed values

petro2061 committed:
changed my sql version

petro2061 committed:
added new test methods

petro2061 committed:
fixed codestyle

petro2061 committed:
added new test class for category controller

petro2061 committed:
fixed codestyle

Elena-Bruyako commented on src/main/java/project/onlinebookstore/service/book/impl/BookServiceImpl.java:
```diff
@@ -31,6 +31,7 @@ public BookDto save(CreateBookRequestDto bookRequestDto) {
         return bookMapper.toBookDto(book);
     }
 
+    @Transactional
```
why did you add this annotation?

Elena-Bruyako commented on src/main/java/project/onlinebookstore/service/book/impl/BookServiceImpl.java:
```diff
@@ -39,6 +40,7 @@ public List<BookDto> findAll(Pageable pageable) {
                 .toList();
     }
 
+    @Transactional
```
why did you add this annotation?

Elena-Bruyako commented on src/test/java/project/onlinebookstore/controller/BookControllerTest.java:
```diff
@@ -0,0 +1,303 @@
+        //Then
+        byte[] jsonResponse = result.getResponse().getContentAsByteArray();
+        BookDto[] bookDtoResponseArray = objectMapper.readValue(jsonResponse, BookDto[].class);
+        List<BookDto> actualBookDtoList = Arrays.stream(bookDtoResponseArray).toList();
```
```suggestion
        List<BookDto> actualBookDtoList = objectMapper.readValue(result.getResponse().getContentAsString(),
                new TypeReference<>() {});
```

Elena-Bruyako commented on src/test/java/project/onlinebookstore/controller/BookControllerTest.java:
```diff
@@ -0,0 +1,303 @@
+        BookDto[] bookDtoResponseArray = objectMapper.readValue(jsonResponse, BookDto[].class);
+        List<BookDto> actualBookDtoList = Arrays.stream(bookDtoResponseArray).toList();
+
+        Assertions.assertFalse(actualBookDtoList.isEmpty());
```
use static imports, check all places

Elena-Bruyako reviewed (changes requested):
see comments

petro2061 committed:
redundant settings have been removed

petro2061 committed:
removed annotation that added before

petro2061 committed:
added static imports

petro2061 commented:
Removed Transactional annotation added before. Also removed reduant properties in application.properties file for tests. Added static imports in test classes.
